### DISCUSSION TITLE
How to make HTTPS requests

### DISCUSSION THREAD

**Discussion by hunterInt** [ionic-team/capacitor#4421] [Q&A]
I'm a Capacitor noob trying to move my existing VueJS project to mobile, however I cannot seem to figure out how to make a HTTPS POST request without getting an error. I'm using `axios` to make the request and the error I get is in the Android Studio log and it shows: 

```
2021-04-01 21:58:26.687 20896-20990/io.foo I/cr_X509Util: Failed to validate the certificate chain, error: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.
2021-04-01 21:58:26.699 20896-21001/io.foo E/chromium: [ERROR:ssl_client_socket_impl.cc(924)] handshake failed; returned -1, SSL error code 1, net_error -202
2021-04-01 21:58:26.718 20896-20896/io.foo E/Capacitor/Console: File: http://localhost/main.ea6aea80.js - Line 79980 - Msg: Error: Request aborted
```

I don't understand what I need to do here. Do I really need to install an SSL certificate on the mobile device in order to make HTTPS requests? If so, where can I get information on how to do that? 

Maybe Capacitor is too "native" for me, I just want to wrap a webapp into a mobile app. But I am probably just missing something here :sweat_smile:

**Comment by jcesarmobile** [comment]
Is it a development server or the production server?
Looks like the certificates are not properly configured.

**Reply by hunterInt** [reply]
It's a staging server. I'm able to make HTTPS requests to it from my browser without issues. I also tried wrapping this project using cordova and was also able to make requests to it without issue, but I'd prefer to use capacitor as it seems cordova itself is a little outdated.

**Reply by jcesarmobile** [reply]
Cordova disables certificate checks in debug builds, so it looks like it works, but then in the release builds it breaks.

**Reply by hunterInt** [reply]
@jcesarmobile oh. So how do I get this to work?

**Reply by jcesarmobile** [reply]
Solving the certificate problems in the server

**Reply by hunterInt** [reply]
@jcesarmobile I could be wrong, but I don't see how this can be a server issue though. I'm able to make secure requests to my server from both my desktop and mobile browsers.

**Comment by davidracl** [comment]
Same problem. I am hosting on my local IIS a project to provide API for a capacitor project, but I have no idea how to apply my self signed certificate to the mobile app.

**Comment by hunterInt** [comment]
UPDATE: This now works with CapacitorJS V3 (but not in V2 still)
